src/externalized/DefaultContentManager.cc

```diff
@@ -203,6 +198,19 @@ DefaultContentManager::DefaultContentManager(RustPointer<EngineOptions> engineOp
 	m_loadingScreenModel = NULL;
 	m_samSitesAirControl = NULL;
 
+	// Initialize temp dir
+	RustPointer<TempDir> tempDir(TempDir_create());
+	if (tempDir.get() == NULL) {
+		RustPointer<char> err{ getRustError() };
+		auto error = ST::format("Failed to create temporary directory: {}", err.get());
+		SLOGE(error);
+		throw std::runtime_error(error.c_str());
```

Comment:
Do we really still not have a function that does both (or all four)?

Reply:
Now we have the termination handler printing exception messages, we might not need the `SLOGE` line.

I have started creating custom exception classes that takes `ST::string` (https://github.com/ja2-stracciatella/ja2-stracciatella/blob/master/src/externalized/Exceptions.h). Let me know your thoughts...

Reply:
I removed the explicit logging here.  :+1: for the custom exception classes.

Currenly it is quite inconsistent which errors are handled through the TerminationHandler, which are catched and logged and which are just ignored with `catch (...)`. Personally I'd say all fatal exceptions should be handled by the TerminationHandler an non-fatal ones should be catched and in most cases logged with warning log level.